docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+Script used to parse the arparse block of all the autometa scripts, copy them to a new
+file and run the help text (--help) from there.
+Used as an alternative to sphinxcontrib-programoutput which used all the heavy dependencies
+preventing the intergration with readthedocs.
+"""
+
+import glob
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import textwrap
+
+
+def get_argparse_block(fpath):
+    """
+    Copies the argparse block of each script in a temporary file
+
+    Parameters
+    ----------
+    fpath : string
+        file path that needs to be documented,
+        e.g. '</path/to/autometa/script.py>'
+
+    Returns
+    -------
+    string
+        path to temporary file that has the argparse block
+        e.g. '</path/to/tmp/file>'
+    """
+    writing = False
+    outlines = ""
+    with open(fpath) as fh:
+        for line in fh:
+            line = line.strip()
+            if line == "import argparse":
+                writing = True
+                imports = ["import os",
+                           "import multiprocessing as mp", ""]
+                # the blank "" is needed to give a new line else error
+                import_lines = "\n".join(imports)
+                outlines += import_lines
+            # Add usage  = ScriptName.py in argparse block
+            if "argparse.ArgumentParser" in line:
+                fname = os.path.basename(fpath)
+                # Uses the starting `(` of Argument.parser, and replaces it with ( usage=
+                usage = f"( usage = '{fname}', "
+                line = line.replace("(", usage)
+            if writing:
+                # regex is used to capture the constant string eg. config.DEFAULT_FPATH
+                # then convert it to "config.DEFAULT_FPATH"
+                # Regex functioning: lower case letter (optional) followed by a '.' (zero or one) upper case (atleast one)
+                # the required '_', upper case (atleast one), then underscore ('_') followed by upper case, both os them optional
+                #captures, config.DEFAULT_FPATH, MARKERS_DIR, wq.WORK_QUEUE_DEFAULT_PORT and DEFAULT_FPATH
+                # add the a comment there and change the variable 'cap_str'
+                cap_str = re.search(r"[a-z]*\.?[A-Z]+_[A-Z]+[_A-Z]*", line)
```

Comment:
It can but then it will also start to match with constants like `logger.DEBUG` and similar ones. I tried this and was getting an error. Is it possible to make a standard way of declaring these constants like mentioning in `template.py` ?

Reply:
Why is there a problem with matching `logger.DEBUG`? What errors are you getting?